add optional "filter" type to image resize function

aaron404 opened the pull request:
**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?

- I have added a new `filter` parameter to the `resize_image` function. It defaults to `lanczos3` which is what was previously hardcoded. I had to add a mirror of the `Filter` type from the `image` crate to impl `Hash` for it. I'm not sure if there's a cleaner way to do this.
- I added a sprite from [pixelfrog](https://pixelfrog-assets.itch.io/tiny-swords) that has public domain usage rights to help demonstrate the differences of the filters in the updated docs.

Keats committed:
Next version

pranitbauva1997 committed:
exclude paginated pages in sitemap (#2555)

This fixes #2527.

aaron404 committed:
add optional "filter" type to image resize function

aaron404 commented:
I'm not sure if I have incorrectly configured something, but it seems that running `cargo test` from the repo root does not execute all the tests that are used by the automated builder. I have a few test cases to fix up

Keats commented:
`cargo test` will only run the root crate, you need to run `cargo test --all` to run tests for all the crates in a workspace

aaron404 committed:
fix tests and build warnings

hegerdes committed:
feat: build relases on gh-actions & support linux arm64 (#2629)

* feat: port zola release build to gh actions & support linux arm64

Signed-off-by: Henrik Gerdes <hegerdes@outlook.de>

* fix: add buildx support for multi-arch docker images

Signed-off-by: Henrik Gerdes <hegerdes@outlook.de>

* fix: pin gh release action to sha & allow pre-releases

Signed-off-by: Henrik Gerdes <hegerdes@outlook.de>

* fix: use env in gh action for linux arm build

Signed-off-by: Henrik Gerdes <hegerdes@outlook.de>

* chore: switch to dtolnay/rust-toolchain action for rust setup

Signed-off-by: Henrik Gerdes <hegerdes@outlook.de>

* fix: windows archive step

Signed-off-by: Henrik Gerdes <hegerdes@outlook.de>

---------

Signed-off-by: Henrik Gerdes <hegerdes@outlook.de>

Keats commented on components/imageproc/src/ops.rs:
```diff
@@ -46,12 +53,21 @@ impl ResizeOperation {
+            "catmullrom" => FilterType::CatmullRom,
+            "gaussian" => FilterType::Gaussian,
+            _ => return Err(anyhow!("Invalid filter type: {}", filter)),
+        };
```
Can we convert directly to the filter type from the image crate and skip our own?

aaron404 replied:
I went with a custom type because FilterType from the image crate is not `Hash`, so it breaks downstream functionality when the processed image paths are generated.

Keats replied:
I've just made a PR to impl it, let's see if it works

Keats replied:
i think it's released in the most recent version

xixishidibei committed:
chore: add missing symbol (#2641)

Signed-off-by: xixishidibei <xixishidibei@outlook.com>

Keats committed:
remove square brackets

SamWilsn committed:
Allow treating a missing highlight language as error (#2642)

bemyak committed:
Move sublime into config module (#2580)

Having a crate depend on files outside of its directory
breaks building Zola as a dependency using Nix

ZzMzaw committed:
Add pagination info to sections (#2653)

* Add pagination info to get_section

* Add documentation for pagination information

* Stick to section properties

`paginated` attributes removed

jamiedumont committed:
Handle file events emitted by Nova on macOS (#2674)

Commit:
Improve accessibility by nesting bottom footnotes inside footer element (#2688)

This adds semantic meaning to the footnotes organization and improves
accessibility by aiding disabled users who rely on assistive devices
such as screen readers which utilize semantic tags such as the footer
element in order to navigate properly.

The hr element is semantically defined as representing a paragraph-level
thematic break. Now that the footnotes are descendants of the footer
element, the hr element originally preceding the footnotes list is
no longer necessary (footnotes are no longer paragraph-level content)
and thus replaced. However, the footer element is given the class
"footnotes" to allow for styling, so the following CSS could be used to
provide a stylistically equivalent visible border separating the
footnotes from the content if so desired:

    .footnotes {
      border-top: 2px groove gray;
    }

Test snapshots are also updated to reflect the new footer elements.

welpo committed:
Add external links class option to Markdown configuration (#2717)

* Add external links class option to Markdown configuration

* Validate external links class

* Rename external link test snapshots

Integral-Tech committed:
refactor: replace static with const for global constants (#2729)

st1020 committed:
feat: do not minify js when minify html

st1020 committed:
test: update tests

Keats committed:
Merge pull request #2732 from st1020/feat/do-not-minify-js-when-minify-html

* feat: do not minify js when minify html

* test: update tests

rparrett committed:
Add optional arg to `serve` for extra paths to watch for changes (#2745)

* Add optional arg to serve for extra paths to watch for changes

* Address feedback

izissise committed:
Add path annotation for codeblock (#2610)

* Add path annotation for codeblock

* Doc

* Rename path to name

Keats committed:
Update changelog

aaron404 committed:
Merge branch 'next' into resize-filters

aaron404 commented:
@Keats I think I have missed something. You mentioned you made a PR to impl it (I assume impl `Hash` for `image`'s `FilterType`, but I wasn't able to find it. Can you point me to it?

Keats commented:
It's https://github.com/image-rs/image/commit/02a4928df9c14bdfa25594facd83391a820b6271
It should be available in recent `image` version

aaron404 commented:
It looks like we also need FilterType to impl `Eq`. I have opened a [PR](https://github.com/image-rs/image/pull/2397) for that

aaron404 commented:
Hmm.. `FilterType` also needs to be `Serialize` and `Deserialize` for use in `EnqueueResponse`. I'm not sure if it makes sense for `image` to impl these traits, but they do use a `serde` cfg feature, so it could be impl'd behind that check. I will submit a PR and see if they are open to it.

Keats commented:
Looks like both PRs have been merged?

Keats commented:
Replaced by https://github.com/getzola/zola/pull/3197
